### PR TITLE
Glue: Hide run queries button in Correlations Page 

### LINKED ISSUE #61030: Glue: Disable Query Editor's "Run Queries" button in the Correlations page
Prometheus Query Editor display "Run Queries" button that ideally should be hidden. It seems it should be possible to set up as it's not shown in Explore:

| Correlations | Explore |
|--|--|
|  Screenshot 2023-01-05 at 11 39 14 | Screenshot 2023-01-05 at 11 38 54 |

### PR DESCRIPTION
**Pull request by ifrost**
When setting up a correlation we don't want to show "Run Queries" button (Loki and Prometheus) as we are not really showing results and the editor is used primarily to write a query.

Fixes #61030

### CODE REVIEW

**Review by ivanahuckova — APPROVED** [review]
LGTM 🙂

**Comment by ifrost** [comment]
Big question to @grafana/dashboards-squad, @grafana/observability-logs and @grafana/observability-metrics - why do we have a "Run Queries" button inside the editor? Would it be possible to break the API in the future and stop providing "runQuery()" handler so the running always happens outside of the editor (related to https://github.com/grafana/grafana/issues/51979)

**Comment by matyax** [comment]
@ifrost the button doesn't need to be there, but the `onRunQuery` prop is shared with the Label Browser, Query Editor, and Query Builder. I can only see the Label Editor being factored out of the component, but the Editor and Builder need it. In the case of the Editor, for example, it's used for the `cmd + shift` and the `unblur` query running behavior.

**Comment by gtk-grafana** [comment]
@ifrost That's a good question, right now there's plenty of usage with `runQuery` within the plugins, so this would require some investigation on a per-datasource basis. Influx doesn't look too bad, but prometheus makes a lot of custom calls to that function.

**Comment by svennergr** [comment]
> Big question to @grafana/dashboards-squad, @grafana/observability-logs and @grafana/observability-metrics - why do we have a "Run Queries" button inside the editor?

Don't know if that answers your question, but I think it's mostly for editing dashboard panels, where you don't have an other `Run Queries` button.

**Review by gelicia — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx [hunk 1]
```diff
@@ -160,23 +186,31 @@ function renderWithMode(mode: QueryEditorMode) {
...
   expect(await screen.findByText('Label filters')).toBeInTheDocument();
 }
 
+function expectRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).toBeInTheDocument();
+}
+
+function expectNoRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).not.toBeInTheDocument();
+}
```
Comment (Elfo404) [inline public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx:212]: let's use more semantic selectors:
```suggestion
function expectRunQueriesButton() {
  expect(screen.getByRole('button', { name: /run queries/i })).toBeInTheDocument();
}

function expectNoRunQueriesButton() {
  expect(screen.queryByRole('button', { name: /run queries/i })).not.toBeInTheDocument();
}
```

File: public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.test.tsx [hunk 2]
```diff
@@ -150,23 +168,31 @@ function renderWithMode(mode: QueryEditorMode) {
...
 }
 
+function expectRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).toBeInTheDocument();
+}
+
+function expectNoRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).not.toBeInTheDocument();
+}
+
```
Comment (Elfo404) [inline public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.test.tsx:195]: same as the other comment:
```suggestion
let's use more semantic selectors:
```suggestion
function expectRunQueriesButton() {
  expect(screen.getByRole('button', { name: /run queries/i })).toBeInTheDocument();
}

function expectNoRunQueriesButton() {
  expect(screen.queryByRole('button', { name: /run queries/i })).not.toBeInTheDocument();
}
```
```

**Review by Elfo404 — CHANGES_REQUESTED** [review]

**Review by Elfo404 — APPROVED** [review]

**Pull request merged**